addons/account/models/account_move.py

```diff
@@ -1748,8 +1748,8 @@ def create_exchange_rate_entry(self, aml_to_fix, amount_diff, diff_in_currency,
             exchange_journal = move.company_id.currency_exchange_journal_id
             aml_model.with_context(check_move_validity=False).create({
                 'name': _('Currency exchange rate difference'),
-                'debit': amount_diff > 0 and aml.amount_residual or 0.0,
-                'credit': amount_diff < 0 and -aml.amount_residual or 0.0,
+                'debit': amount_diff > 0 and abs(aml.amount_residual) or 0.0,
+                'credit': amount_diff < 0 and abs(aml.amount_residual) or 0.0,
```

Comment:
I'm pretty sure this is a very bad idea. Let me some time to investigate plz

Reply:
you should write a test for that particular use case and imo the fix should be more in the lines of what is currently written in v13.0:

https://github.com/odoo/odoo/blob/13.0/addons/account/models/account_move.py#L4247
Instead of using amount_diff, check on amount_residual < 0 (or > 0)

Also this fix should only be forward ported up until we changed the amount_diff to amount_residual (don't remember which version it was)